Add Search bar and related info. (#12)
chamaeleonidae/api: docs/apis/search.md

# Search bar

**Add a Search bar to your product that allows your Users to search your Product, Help center, and any custom content.**

> > **You have content within your product even if you don't think of it that way, any database records that your Users would want to find is a piece of content, akin to a Google search result. Send this content to Chameleon and we will surface it to your Users in the Chameleon search bar. The content can be scoped to a specific [Company](apis/company.md), be recommended when the Search bar is opened and can perform any action from clicking an item.**.

Reviewer: ```suggestion
> **You have content within your product, even if you don't think of it that way. Any database records that your Users would want to find is a piece of content, akin to a Google search result. Send this content to Chameleon and we will surface it to your Users in the Chameleon search bar. The content can be scoped to a specific [Company](apis/company.md), be recommended when the Search bar is opened and can perform any action from clicking an item.**.
```

------

See the specific APIs for more information on how to use Search your product.

 - [REST API](apis/search.md?id=search-rest-api)
 - [JavaScript API](apis/search.md?id=search-js-api)

#### To get Search up and running, complete these items:

1. Create a [Search bar](apis/search.md?id=schema), configure it with `placeholder` etc.
2. Add your Help center as a [`SearchGroup`](apis/search.md?id=schema-search-group).
3. Add the top-level navigation within your product; First download this CSV [template](apis/search.md?id=search-imports-template) or [example](apis/search.md?id=search-imports-example) then [Import](apis/search.md?id=schema-search-imports) it.
4. [Optional] Add specific content from your database
5. Publish your Search bar

> 6. Visit your product where Chameleon is installed and hit `CMD` + `k` (or how you configured `key_meta` and `key_uid`).

Reviewer: ```suggestion
6. Visit your product where Chameleon is installed and hit `CMD` + `k` for MacOs or 'Ctrl" + 'k' for Windows and Linux  (or how you configured `key_meta` and `key_uid`).
```

```json
{
  "search": {
    "id": "5f3c4232c712de665632a6d9",
    "placeholder": "Search for help...",
    "key_meta": true,
    "key_uid": "k",
    "title_recent": "Recently clicked",
    "title_suggestions": "Recommended",
    "title_results": "Your Results",
    "published_at": "2029-04-07T12:18:00Z",
    "groups": [
      {
        "kind": "public",
        "title": "Help center",
        "url": "https://help.your-product.com",
        "icon": {
          "kind": "uid",
          "uid": "book_open"
        }
      },
      {
        "kind": "public",
        "title": "Best practices",
        "url": "https://blog.your-product.com",
        "icon": {
          "kind": "uid",
          "uid": "film"
        }
      }
    ],
    "label_themes": [
      {
        "id": "6f3c4232c712de632a6d9656",
        "style": "filled",
        "style_color_text": "FFF",
        "style_color_background": "FF6666"
      },
      {
        "id": "62c73c423f12de632a6d96fa",
        "style": "filled",
        "style_color_text": "FFF",
        "style_color_background": "66FF66"
      }
    ]
  }
}
```

------

## `SearchItem` Schema :id=schema-search-items

> A search item is a discrete unit of searchable content akin to a Google search result. You have content within your product even if you don't think of it that way, any database records that your Users would want to find is a piece of content. Sent us this content to surface it to your Users in the Chameleon search bar

Reviewer 2: ```suggestion
A search item is a discrete unit of searchable content akin to a Google search result. You have content within your product even if you don't think of it that way, any database records that your Users would want to find is a piece of content. Send us this content to surface it to your Users in the Chameleon search bar
```

> A search item is a discrete unit of searchable content akin to a Google search result. You have content within your product even if you don't think of it that way, any database records that your Users would want to find is a piece of content. Sent us this content to surface it to your Users in the Chameleon search bar

Reviewer: ```suggestion
A search item is a discrete unit of searchable content akin to a Google search result. You have content within your product, even if you don't think of it that way. Any database records that your Users would want to find is a piece of content. Send us this content to surface it to your Users in the Chameleon search bar
```

See the [Type definitions below ↓](apis/search.md?id=search-js-types)

##### Recommended

> When using React, add `<ChmlnHook>` at the same level as your Router => https://www.npmjs.com/package/@chamaeleonidae/chmln

```javascript

chmln.on('app:navigate', (opts: NavigateOpts) => {
  /*
  Nagivate with React router / Vue router / pushState / call window.open / etc.

  Frameworks:
    - React info here => https://www.npmjs.com/package/@chamaeleonidae/chmln
    - Add more frameworks with a PR to this file and receive a $25 gift card

  opts.to is the url configured in the SearchAction configured with `kind=navigate`

  */
});
```

##### Optional

```javascript

chmln.on('cmd:search', (opts: SearchOpts, ctx: Context) => {
  /*
  Optional:
    - Chameleon will handle search for you
    - If you need to add custom "Search" logic use this event to pass back the right items, two options:
      - directly return an array<SearchItem>
      - return a Promise that will resolve with the first argument being an array<SearchItem>.

>     - Best used when you already use Algolia or have a custo msearch endpoint internally

Reviewer 2: ```suggestion
    - Best used when you already use Algolia or have a custom search endpoint internally
```
